packages/tpetra/core/src/Tpetra_Core.cpp

```diff
@@ -313,7 +313,7 @@ namespace Tpetra {
 
     // Tpetra should only finalize Kokkos if it initialized Kokkos.
     // See Github Issue #434.
-    if (tpetraInitializedKokkos_) {
+    if (tpetraInitializedKokkos_ && !Kokkos::is_finalized()) {
```

Comment:
Why did you change?  Who would call `Kokkos::finalize` if Tpetra initialized Kokkos?

Reply:
ping

Reply:
Hopefully nobody.  But it doesn't hurt to be safe.

Reply:
I think this is semantically incorrect.  If you initialized Kokkos you are responsible for finalizing.
If someone else called `finalize` then there is a bug.  The change proposed here hide the bug they don't fix it.